tests/Fixer/PhpUnit/PhpUnitDataProviderReturnTypeFixerTest.php

```diff
@@ -0,0 +1,246 @@
+    public function provideFooCases2(): iterable {}
+    public function provideFooCases3(): iterable {}
+    public function someFunction() {}
+    public function someFunction2() {}
+}',
+            '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider provideFooCases
+     * @dataProvider provideFooCases2
+     */
+    public function testFoo()
+    {
+        /**
+         * @dataProvider someFunction
+         */
+        $foo = /** foo */ function ($x) { return $x + 1; };
+        /**
+         * @dataProvider someFunction2
+         */
+        /* foo */someFunction2();
+    }
+    /**
+     * @dataProvider provideFooCases3
+     */
+    public function testBar() {}
+
+    public function provideFooCases() {}
+    public function provideFooCases2() {}
+    public function provideFooCases3() {}
+    public function someFunction() {}
+    public function someFunction2() {}
+}',
+        ];
+
+        foreach (['abstract', 'final', 'private', 'protected', 'static', '/* private */'] as $modifier) {
+            yield sprintf('test function with %s modifier', $modifier) => [
+                sprintf('<?php
+                    abstract class FooTest extends TestCase {
+                        /**
+                         * @dataProvider provideFooCases
+                         */
+                        %s function testFoo() %s
+                        public function provideFooCases(): iterable {}
+                    }
+                ', $modifier, 'abstract' === $modifier ? ';' : '{}'),
+                sprintf('<?php
+                    abstract class FooTest extends TestCase {
+                        /**
+                         * @dataProvider provideFooCases
+                         */
+                        %s function testFoo() %s
+                        public function provideFooCases() {}
+                    }
+                ', $modifier, 'abstract' === $modifier ? ';' : '{}'),
+            ];
+        }
+    }
+
+    /**
+     * @requires PHP ^7.4
+     *
+     * @dataProvider provideFix7Cases
+     */
+    public function testFix7(string $expected, ?string $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return iterable<array{string, string}>
+     */
+    public static function provideFix7Cases(): iterable
+    {
+        yield 'data provider with return type namespaced class starting with iterable' => self::mapToTemplate(
+            ': iterable',
+            ': iterable \ Foo',
+        );
+
+        yield 'data provider with return type namespaced class and comments' => self::mapToTemplate(
+            ': iterable',
+            ': Foo/* Some info */\/* More info */Bar',
+        );
+    }
+
+    /**
+     * @return array{string, string}
+     */
+    private static function mapToTemplate(string $expected, string $actual): array
+    {
+        return [
+            sprintf(self::TEMPLATE, $expected, $expected),
```

Comment:
With positional placeholders in the template (`%1$s`) `$expected` could be passed only once.